File: guides/micronaut-views-thymeleaf/java/src/main/java/example/micronaut/PhotosClient.java
package example.micronaut;

import io.micronaut.core.annotation.Blocking;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.PathVariable;
import io.micronaut.http.client.annotation.Client;

@Client(id = "photos") // <1>
public interface PhotosClient {

    @Get("/photos/{id}")// <2>
    @Blocking
    Photo findById(@PathVariable Long id); // <3>
}

File: guides/micronaut-views-thymeleaf/python/src/example/micronaut/photos_client.py
from abc import ABC, abstractmethod
from typing import Annotated

from micronaut.core.annotation import Blocking
from micronaut.http.annotation import Get, PathVariable
from micronaut.http.client.annotation import Client

from .photo import Photo


@Client(id="photos")  # <1>
class PhotosClient(ABC):

    @Get("/photos/{id}")  # <2>
    @Blocking
    @abstractmethod
    def find_by_id(self, id: Annotated[int, PathVariable]) -> Photo:  # <3>
        ...
